src/recommendationservice/recommendation_server.py

```diff
@@ -60,16 +63,35 @@ def Watch(self, request, context):
 
 
 def get_product_list(request_product_ids):
+    global first_run
+    global cached_ids
     with tracer.start_as_current_span("get_product_list") as span:
         max_responses = 5
 
         # Formulate the list of characters to list of strings
         request_product_ids_str = ''.join(request_product_ids)
         request_product_ids = request_product_ids_str.split(',')
 
-        # Fetch list of products from product catalog stub
-        cat_response = product_catalog_stub.ListProducts(demo_pb2.Empty())
-        product_ids = [x.id for x in cat_response.products]
+        # Feature flag scenario - Cache Leak
+        if check_feature_flag("recommendationCache"):
+            if random.random() < 0.219 or first_run:
+                first_run = False
+                span.set_attribute("app.cache_hit", False)
```

Comment:
We need to add `app.cache_hit` to: https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/manual_span_attributes.md#recommendationservice

Reply:
done